regression-test/suites/nereids_rules_p0/infer_predicate/pull_up_predicate_set_op.groovy

```diff
@@ -0,0 +1,215 @@
+    """
+
+    sql """
+    insert into test_pull_up_predicate_set_op3 values(1,'d2',2,2),(-2,'d2',2,2),(0,'d2',3,5);
+    """
+
+    qt_intersect """    
+    explain shape plan
+    select * from      (select a,b from test_pull_up_predicate_set_op1 where a<1 intersect select a,b from test_pull_up_predicate_set_op2 where b>'ab') t inner join test_pull_up_predicate_set_op3 t3
+    on t3.a=t.a and t3.b=t.b;
+    """
+    qt_except """    
+    explain shape plan
+    select * from      (select a,b from test_pull_up_predicate_set_op1 where a<1 except select a,b from test_pull_up_predicate_set_op2 where b>'ab') t inner join test_pull_up_predicate_set_op3 t3
+    on t3.a=t.a and t3.b=t.b;
+    """
+    qt_union """    
+    explain shape plan
+    select * from      (select a,b from test_pull_up_predicate_set_op1 where a<1 union select a,b from test_pull_up_predicate_set_op2 where a<1) t inner join test_pull_up_predicate_set_op3 t3
+    on t3.a=t.a and t3.b=t.b;
+    """
+
+    qt_intersect_res """    
+    select t.a,t3.b from      (select a,b from test_pull_up_predicate_set_op1 where a<1 intersect select a,b from test_pull_up_predicate_set_op2 where b>'ab') t inner join test_pull_up_predicate_set_op3 t3
+    on t3.a=t.a and t3.b=t.b order by 1,2;
+    """
+    qt_except_res """    
+    select t.a,t3.b from      (select a,b from test_pull_up_predicate_set_op1 where a<1 except select a,b from test_pull_up_predicate_set_op2 where b>'ab') t inner join test_pull_up_predicate_set_op3 t3
+    on t3.a=t.a and t3.b=t.b  order by 1,2;
+    """
+    qt_union_res """    
+    select t.a,t3.b from      (select a,b from test_pull_up_predicate_set_op1 where a<1 union select a,b from test_pull_up_predicate_set_op2 where a<1) t inner join test_pull_up_predicate_set_op3 t3
+    on t3.a=t.a and t3.b=t.b  order by 1,2; """
+
+    qt_intersect_one_side """
+    explain shape plan 
+    select t.a,t3.b from      (select a,b from test_pull_up_predicate_set_op1 intersect select a,b from test_pull_up_predicate_set_op2 where b>'ab') t inner join test_pull_up_predicate_set_op3 t3 
+    on t3.a=t.a and t3.b=t.b;
+    """
+
+    qt_intersect_one_side_no_filter """
+    explain shape plan 
+    select t.a,t3.b from      (select a,b from test_pull_up_predicate_set_op1 intersect select a,b from test_pull_up_predicate_set_op2 ) t inner join test_pull_up_predicate_set_op3 t3 
+    on t3.a=t.a and t3.b=t.b;
+    """
+
+    qt_except_first_no_filter """
+    explain shape plan 
+    select t.a,t3.b from      (select a,b from test_pull_up_predicate_set_op1 except select a,b from test_pull_up_predicate_set_op2 where b>'ab') t inner join test_pull_up_predicate_set_op3 t3 
+    on t3.a=t.a and t3.b=t.b;
+    """
+    qt_except_second_no_filter """
+    explain shape plan 
+    select t.a,t3.b from      (select a,b from test_pull_up_predicate_set_op1 where a>1 except select a,b from test_pull_up_predicate_set_op2 ) t inner join test_pull_up_predicate_set_op3 t3 
```

Comment:
add case for pulling the filter or the constant condition up with non-inner join?